openassessment/assessment/test/test_staff.py

```diff
@@ -0,0 +1,342 @@
+        if after_type == 'staff':
+            return
+
+        # Create assessment
+        tim_sub, tim = self._create_student_and_submission("Tim", "Tim's answer", override_steps=[after_type])
+
+        staff_score = "few"
+        # Staff assess it
+        staff_assessment = staff_api.create_assessment(
+            tim_sub["uuid"],
+            "Dumbledore",
+            OPTIONS_SELECTED_DICT[staff_score]['options'], dict(), "",
+            RUBRIC,
+        )
+        # Keep the workflow updated
+        workflow_api.update_from_assessments(tim_sub["uuid"], self.STEP_REQUIREMENTS)
+
+        # Verify both assessment and workflow report correct score
+        self.assertEqual(staff_assessment["points_earned"], OPTIONS_SELECTED_DICT[staff_score]["expected_points"])
+        workflow = workflow_api.get_workflow_for_submission(tim_sub["uuid"], self.STEP_REQUIREMENTS)
+        self.assertEqual(workflow["score"]["points_earned"], OPTIONS_SELECTED_DICT[staff_score]["expected_points"])
+
+        # Now, non-force asses with a 'most' value
+        # This was selected to match the value that the ai test will set
+        unscored_assessment = OPTIONS_SELECTED_DICT["most"]
+        assessment = after_assess(tim_sub["uuid"], tim["student_id"], unscored_assessment["options"])
+        # and update workflow with new scores
+        requirements = self.STEP_REQUIREMENTS
+        if after_type == 'peer':
+            requirements = {"peer": {"must_grade": 0, "must_be_graded_by": 1}}
+        workflow_api.update_from_assessments(tim_sub["uuid"], requirements)
+
+        # Verify both assessment and workflow report correct score (workflow should report previous value)
+        self.assertEqual(assessment["points_earned"], unscored_assessment["expected_points"])
+        workflow = workflow_api.get_workflow_for_submission(tim_sub["uuid"], self.STEP_REQUIREMENTS)
+        self.assertEqual(workflow["score"]["points_earned"], OPTIONS_SELECTED_DICT[staff_score]["expected_points"])
+
+    def test_invalid_rubric_exception(self):
+        # Create a submission
+        tim_sub, tim = self._create_student_and_submission("Tim", "Tim's answer")
+
+        # Define invalid rubric
+        invalid_rubric = copy.deepcopy(RUBRIC)
+        for criterion in invalid_rubric["criteria"]:
+            for option in criterion["options"]:
+                option["points"] = -1
+
+        # Try to staff assess with invalid rubric
+        with self.assertRaises(StaffAssessmentRequestError) as context_manager:
+            staff_assessment = staff_api.create_assessment(
+                tim_sub["uuid"],
+                "Dumbledore",
+                OPTIONS_SELECTED_DICT["most"]["options"], dict(), "",
+                invalid_rubric,
+            )
+        self.assertEqual(str(context_manager.exception), u"Rubric definition was not valid")
+
+    def test_invalid_rubric_options_exception(self):
```

Comment:
I'd still like this to be DDT test with a variety of different invalid options. I think it would be clearer that all of the validation is happening, and it would be more extensible.